executor/common_kvm_arm64.h

```diff
@@ -66,13 +69,21 @@ static volatile long syz_kvm_setup_cpu(volatile long a0, volatile long a1, volat
 	}
 
 	struct kvm_vcpu_init init;
-	ioctl(cpufd, KVM_ARM_PREFERRED_TARGET, &init);
+	// Queries KVM for preferred CPU target type.
+	ioctl(vmfd, KVM_ARM_PREFERRED_TARGET, &init);
 	init.features[0] = features;
+	// Use the modified struct kvm_vcpu_init to initialize the virtual CPU.
 	ioctl(cpufd, KVM_ARM_VCPU_INIT, &init);
 
 	if (text_size > 1000)
 		text_size = 1000;
 	memcpy(host_mem, text, text_size);
+	memcpy(host_mem + text_size, BRK_INSTR, sizeof(BRK_INSTR));
+
+	struct kvm_guest_debug debug = {
+	    .control = KVM_GUESTDBG_ENABLE,
+	};
+	ioctl(cpufd, KVM_SET_GUEST_DEBUG, &debug);
```

Comment:
What happens w/o this? How does it handle BRK? I would assume it will still exit, what else can it do?

Reply:
I found unit tests for different modes very useful when worked on x86 kvm:
https://github.com/google/syzkaller/blob/master/executor/test_linux.h
Without them it's almost impossible to ensure that something and haven't broke, esp when we get to more complex modes, e.g. nested virt.

Maybe it's possible to write them as `sys/linux/text/*` unit tests instead. We just need to figure out how to extract and check register contents to ensure that the code actually executed. But then these will serve as good fuzzer seeds as well.